pkg/Cpanel/Security/Advisor/Assessors/Mysql.pm

```diff
@@ -165,25 +165,41 @@ sub _check_for_public_bind_address {
         else {
             $self->add_bad_advice(
                 text       => "The MySQL service is currently configured to listen on a public address: (bind-address=$bind_address)",
-                suggestion => 'Configure bind-address=127.0.0.1 in /etc/my.cnf',
+                suggestion => [
+                    'Configure bind-address=127.0.0.1 in /etc/my.cnf, or close port [_1] in the server’s firewall.',
+                    $port
+                ],
             );
         }
     }
     else {
-        if ( @deny_rules && @deny_rules_6 ) {
+        if ( ( @deny_rules && @deny_rules_6 ) || ( csf_port_closed($port) ) ) {
             $self->add_good_advice( text => "The MySQL port is blocked by the firewall, effectively allowing only local connections." );
         }
         else {
             $self->add_bad_advice(
                 text       => 'The MySQL service is currently configured to listen on all interfaces: (bind-address=*)',
-                suggestion => 'Configure bind-address=127.0.0.1 in /etc/my.cnf',
+                suggestion => [
+                    'Configure bind-address=127.0.0.1 in /etc/my.cnf, or close port [_1] in the server’s firewall.',
+                    $port
+                ],
             );
         }
     }
 
     return 1;
 }
 
+sub csf_port_closed {
```

Comment:
Actually, I found another error.  If the file doesn't exist, the port is said to be blocked.  The code should be:

```perl
sub csf_port_closed {
    my ($port) = @_;
    my $contains = config_key_contains_port( '/etc/csf/csf.conf', 'TCP_IN', $port );
    return if !defined $contains;
    return !$contains;
}
```

That said, you are welcome to copy&paste my code snippets into code that you are submitting to this project; no need for attribution (after all, the pull request has this information anyway).